### DISCUSSION TITLE
Get number of mutations in CDR3 region

### DISCUSSION THREAD

**Discussion by Blowfish82** [milaboratory/mixcr#992] [Q&A] [answered]
Dear Team,

I want to know the number of mutations per clonotype in the CDR3 region of my IGH data.
Is there a way to see the number of mutations in one of the export formats? 
I usually work with the tables resulting from command "exportClones".

Your reply would be extremely helpful.

**Comment by PoslavskySV** [comment]
Hi, 

could you elaborate in more detail. Do you mean the number of mutations in germline parts of CDR3 region?

**Comment by Blowfish82** [comment, marked as answer]
Thank you for your fast response,

Yes, you are right, I mean the germline part. Let´s look at the Vgene, 
I think the colum "vBestIdentityPercent" in the exportClones format reflects a length normalized mutation rate to the best aligned vGene.
If I also want to know, how many (absolute, not percentage) mutations were detected in the V-part of the sequence (not as percentage), in which column should I look for it?

Many thanks!

**Reply by Blowfish82** [reply]
I think I got the solution. 
In the "exportClones" Table, there is a column called "allVAlignments".
There all VHits and their alignment statistics are listet. It has a pattern as follows, segmented by "|":


60|304|325|0|244|SA180TSA194GSA209GSA213TSG242CSG254CSC260TST261ASG275ASC293TSG297ASA299G|1052.0

The last entry is the aligment score (here it is 1052), and the second last shows all mutations (e.g. substitutions etc.) and at which positions they occur. 
So If I count all  numbers between letters (grep"[A-Z][0-9]") then I get the number of mutations (its 12 mutations in this case).

Think my problem is solved.
Have a nice day.

**Reply by PoslavskySV** [reply]
Hi,

You are right!  `vBestIdentityPercent` reflects the number of mutations normalized by the length of V part of clonotype [assembling feature](https://docs.milaboratories.com/mixcr/reference/mixcr-assemble/) (i.e. if you assemble clones by CDR3 that is [`GermlineVCDR3Part`](https://docs.milaboratories.com/mixcr/reference/ref-gene-features/#germline-features), if assembling feature is `VDJRegion` than it is `VRegion`).

For the number of mutations  you are also right. You can check [this doc](https://docs.milaboratories.com/mixcr/reference/ref-mutations-encoding/) for the details about encoding. 

Also, we'll add the number of mutations in export in upcoming release - thank you for reporting!

**Reply by amikelov** [reply]
I also would like to mention the possibility to calculate the number of mutations in CDR3 for the clonotypes which are part of lineage trees - for them we are able to reconstruct germline CDR3 sequences with N-nucleotides and so can describe mutations even in these regions.
Take a look at our guide for BCR lineage tracing, you might find it helpful:
https://docs.milaboratories.com/mixcr/guides/b-cell-lineages-webinar/
